#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_dont_warn_when_only_checking.dpatch by  <gwolf@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Turns off the warnings reporting when the user requests to check whether a file
## DP: is or not part of the archive. (see bug #318497)

@DPATCH@
diff -urNad trunk~/lib/Archive/Tar.pm trunk/lib/Archive/Tar.pm
--- trunk~/lib/Archive/Tar.pm	2007-05-14 10:32:03.000000000 -0500
+++ trunk/lib/Archive/Tar.pm	2007-05-14 10:42:23.000000000 -0500
@@ -408,7 +408,10 @@
     my $self = shift;
     my $full = shift or return;
 
+    my $tmp_warn = $WARN;
+    $WARN = 0;
     return 1 if $self->_find_entry($full);
+    $WARN = $tmp_warn;
     return;
 }
 
